frame/authority-discovery/src/lib.rs

```diff
@@ -75,31 +87,36 @@ impl<T: Config> Pallet<T> {
 	/// Retrieve authority identifiers of the current and next authority set
 	/// sorted and deduplicated.
 	pub fn authorities() -> Vec<AuthorityId> {
-		let mut keys = Keys::<T>::get();
-		let next = NextKeys::<T>::get();
+		let mut keys = Keys::<T>::get().to_vec();
+		let next = NextKeys::<T>::get().to_vec();
 
 		keys.extend(next);
 		keys.sort();
 		keys.dedup();
 
-		keys
+		keys.to_vec()
 	}
 
 	/// Retrieve authority identifiers of the current authority set in the original order.
-	pub fn current_authorities() -> Vec<AuthorityId> {
+	pub fn current_authorities() -> BoundedVec<AuthorityId, T::MaxAuthorities> {
 		Keys::<T>::get()
 	}
 
 	/// Retrieve authority identifiers of the next authority set in the original order.
-	pub fn next_authorities() -> Vec<AuthorityId> {
+	pub fn next_authorities() -> BoundedVec<AuthorityId, T::MaxAuthorities> {
 		NextKeys::<T>::get()
 	}
 
-	fn initialize_keys(keys: &[AuthorityId]) {
+	fn initialize_keys(keys: &Vec<AuthorityId>) {
 		if !keys.is_empty() {
 			assert!(Keys::<T>::get().is_empty(), "Keys are already initialized!");
-			Keys::<T>::put(keys);
-			NextKeys::<T>::put(keys);
+
+			let bounded_keys =
+				BoundedVec::<AuthorityId, T::MaxAuthorities>::try_from((*keys).clone())
+					.expect("Keys vec too big");
+
+			Keys::<T>::put(bounded_keys.clone());
```

Comment:
we can use a reference instead to avoid cloning
```suggestion
			Keys::<T>::put(&bounded_keys);
```